dotnet/src/SemanticKernel.Abstractions/Search/ITextSearchStringMapper.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Interface for mapping between a text search data model, and a string value.
+/// </summary>
+[Experimental("SKEXP0001")]
+public interface ITextSearchStringMapper
+{
+    /// <summary>
+    /// Map from the text search data result to a string value.
+    /// </summary>
+    /// <param name="result">The instance of the text search result to map.</param>
+    /// <returns>A string value.</returns>
+    string MapFromResultToString(object result);
```

Comment:
Same comment regarding using generics here over `object` as described in previous comment. Also, I'm wondering, do we need this mapper at all, since the result of this mapper is always a `string`, but each object can override its `ToString` method and return a string representation of itself. On the other hand, it's going to be unavailable when we want to map a class to a `string`, but we don't have access to this class, and we can't extend it because it's `sealed`. But even in this scenario, if most of the time JSON serialization will be performed to get a string representation of an object (in this PR I can see `JsonSerializer.Serialize(result)`), then maybe we can do that by default, but allow users to pass `JsonSerializerOptions`, so they can adjust serialization process based on their needs.

Reply:
We're not using generic in `ITextSearch` so introducing it for the mappers feels awkward. I have updated the API documentation to better match the terminology used elsewhere.